include/gz/common/Plugin.hh

```diff
@@ -0,0 +1,196 @@
+/*
```

Comment:
We're deprecating all the `*Plugin*` files from `ign-common` on Garden, in favor of using the `ign-plugin` library. So I think we don't need to add `gz` counterparts to it.

I'd be ok doing this in a separate PR, we don't even have `IGN_DEPRECATED` notices on them anyway.

Reply:
Let's do this in a separate PR, so it can be tracked independently.

I'll track this in the issue here: https://github.com/ignitionrobotics/ign-common/issues/349
But I suspect I'll be focused on the other libraries.